In [ ]:
import pandas as pd
import numpy as np

In [ ]:
# Membuat list berisi data transaksi
transaction_amounts = [100, 150, 50, 100, 130, 120, 100, 85, 70, 150, 150, 120, 50, 100, 100, 140, 90, 150, 50, 90, 120, 100, 110, 75, 65]

# Membuat data frame dari list yang telah disusun
TransactionAmount = pd.DataFrame({'Transaction Amount': transaction_amounts})

# menampilkan atau print data frame tersebut
print(TransactionAmount )
TransactionAmount.info()


    Transaction Amount
0                  100
1                  150
2                   50
3                  100
4                  130
5                  120
6                  100
7                   85
8                   70
9                  150
10                 150
11                 120
12                  50
13                 100
14                 100
15                 140
16                  90
17                 150
18                  50
19                  90
20                 120
21                 100
22                 110
23                  75
24                  65
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25 entries, 0 to 24
Data columns (total 1 columns):
 #   Column              Non-Null Count  Dtype
---  ------              --------------  -----
 0   Transaction Amount  25 non-null     int64
dtypes: int64(1)
memory usage: 328.0 bytes


In [ ]:
# Menghitung measures of central tendency
rataan = TransactionAmount['Transaction Amount'].mean()
median = TransactionAmount['Transaction Amount'].median()
modus = TransactionAmount['Transaction Amount'].mode()
q1 = TransactionAmount['Transaction Amount'].quantile(0.25)
q3 = TransactionAmount['Transaction Amount'].quantile(0.75)

# Menampilkan hasil perhitungan
print(f"Mean = {rataan}")
print(f"Median = {median}")
print(f"Modus = {modus}")
print(f"q1 = {q1}")
print(f"q3 = {q3}")

Mean = 102.6
Median = 100.0
Modus = 0    100
Name: Transaction Amount, dtype: int64
q1 = 85.0
q3 = 120.0


# Perhitungan Measures of variability

In [ ]:
# Menghitung measures of variability
interval_data = TransactionAmount['Transaction Amount'].max() - TransactionAmount['Transaction Amount'].min()
ragam = TransactionAmount['Transaction Amount'].var()
std_deviasi = TransactionAmount['Transaction Amount'].std()

# menampilkan hasil perhitungan
print(f"Range = {interval_data}")
print(f"Variance = {ragam}")
print(f"Standard Deviation = {std_deviasi}")


Range = 100
Variance = 1012.7499999999999
Standard Deviation = 31.823733281939123


# Uji Hipotesis pada dataset

In [ ]:
import scipy.stats as stats

# mensetting uji hipotesis berdasarkan rata-rata
# H0 : Rata-rata transaksi sama 100
# H1 : Rata-rata transaksi tidak sama dengan 100
tes = 100

# Melakukan uji hipotesis
t_stat, p_value = stats.ttest_1samp(TransactionAmount, tes)

# Menampilkan hasil perhitungan uji hipotesis
print(f"T-statistic: {t_stat}")
print(f"P-value: {p_value}")

# Melakukan cek untuk uji tes hipotesis pada alpha (e.g., 0.05)
alpha < 0.05
if p_value <= alpha:
    print("Rata-rata transaksi sama dengan 100")
else:
    print("Rata-rata transaksi tidak sama dengan 100")


T-statistic: [0.40850016]
P-value: [0.68652848]
Rata-rata transaksi tidak sama dengan 100


# **Interpretasi Hasil**

Nilai rata-rata pada 25 transaksi keuangan sebesar 102.6 dengan nilai transaksi paling banyak dilakukan sebesar 100. Selain itu, nilai transaksi paling tinggi (maksimal) sebesar 150 dan terendahnya sebesar (minimal) sebesar 50.



Adapun ragam (variance) transaksi pada data tersebut hampir mendekati nilai rata-ratanya sehingga dapat dikatakan bahwa nilai transaksi pada data tersebut berada pada nilai transaksi sebesar 102.6, atau menyebar diantara nilaiu rata-rata transaksi.  

Dari hasil uji hipotesis terhadap nilai tes rata-rata transaksi sebesar 100 membuktikan tolak H0. Dengan kata lain bahwa nilai tersebut tidak sama 100.